core/indexer/process/transactions/transactionsProcessor.go

```diff
@@ -0,0 +1,239 @@
+
+	tdp.addTxsLogsIfNeeded(transactions)
+
+	txsSlice := append(convertMapTxsToSlice(transactions), rewardsTxs...)
+
+	return txsSlice, dbSCResults, dbReceipts, alteredAddresses
+}
+
+func (tdp *txDatabaseProcessor) setStatusOfTxsWithSCRS(
+	transactions map[string]*types.Transaction,
+	countScResults map[string]int,
+) {
+	for hash, nrScResult := range countScResults {
+		tx, ok := transactions[hash]
+		if !ok {
+			continue
+		}
+
+		tx.HasSCR = true
+
+		if isRelayedTx(tx) {
+			tx.GasUsed = tx.GasLimit
+			fee := tdp.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, tx.GasUsed)
+			tx.Fee = fee.String()
+
+			continue
+		}
+
+		if nrScResult < minimumNumberOfSmartContractResults {
+			if len(tx.SmartContractResults) > 0 {
+				scResultData := tx.SmartContractResults[0].Data
+				if isScResultSuccessful(scResultData) {
+					// ESDT contract calls generate just one smart contract result
+					continue
+				}
+			}
+
+			tx.Status = transaction.TxStatusFail.String()
+
+			tx.GasUsed = tx.GasLimit
+			fee := tdp.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, tx.GasUsed)
+			tx.Fee = fee.String()
+		}
+	}
+}
+
+func (tdp *txDatabaseProcessor) iterateSCRSAndConvert(
+	txPool map[string]data.TransactionHandler,
+	header data.HeaderHandler,
+	transactions map[string]*types.Transaction,
+) ([]*types.ScResult, map[string]int) {
+	//we can not iterate smart contract results directly on the miniblocks contained in the block body
+	// as some miniblocks might be missing. Example: intra-shard miniblock that holds smart contract results
+	scResults := groupSmartContractResults(txPool)
+
+	dbSCResults := make([]*types.ScResult, 0)
+	countScResults := make(map[string]int)
+	for scHash, scResult := range scResults {
+		dbScResult := tdp.txBuilder.convertScResultInDatabaseScr(scHash, scResult, header)
+		dbSCResults = append(dbSCResults, dbScResult)
+
+		tx, ok := transactions[string(scResult.OriginalTxHash)]
+		if !ok {
+			continue
+		}
+
+		tx = tdp.addScResultInfoInTx(dbScResult, tx)
+		countScResults[string(scResult.OriginalTxHash)]++
+		delete(scResults, scHash)
+
+		// append child smart contract results
+		scrs := findAllChildScrResults(scHash, scResults)
+		for childScHash, sc := range scrs {
+			childDBScResult := tdp.txBuilder.convertScResultInDatabaseScr(childScHash, sc, header)
+
+			tx = tdp.addScResultInfoInTx(childDBScResult, tx)
+			countScResults[string(scResult.OriginalTxHash)]++
+		}
+	}
+
+	return dbSCResults, countScResults
+}
+
+func findAllChildScrResults(hash string, scrs map[string]*smartContractResult.SmartContractResult) map[string]*smartContractResult.SmartContractResult {
+	scrResults := make(map[string]*smartContractResult.SmartContractResult)
+	for scrHash, scr := range scrs {
+		if string(scr.OriginalTxHash) == hash {
+			scrResults[scrHash] = scr
+			delete(scrs, scrHash)
+		}
+	}
+
+	return scrResults
+}
+
+func (tdp *txDatabaseProcessor) addTxsLogsIfNeeded(txs map[string]*types.Transaction) {
+	defer tdp.txLogsProcessor.Clean()
```

Comment:
Another one: the function is called addTxsLogsIfNeeded but calls a subcomponent's clean function.

Reply:
refactored.